.github/workflows/build.yaml

```diff
@@ -644,47 +644,62 @@ jobs:
 
   scan-images-with-roxctl:
     if: github.event_name == 'push'
-    runs-on: ubuntu-latest
     needs:
       - build-and-push-main
-    container:
-      image: quay.io/stackrox-io/apollo-ci:stackrox-test-0.3.68
-      env:
-        STACKROX_CI_INSTANCE_API_KEY: ${{ secrets.STACKROX_CI_INSTANCE_API_KEY }}
-        STACKROX_CI_INSTANCE_CENTRAL_HOST: ${{ secrets.STACKROX_CI_INSTANCE_CENTRAL_HOST }}
+    name: Check images for vulnerabilities
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed for stackrox/central-login to create the JWT token.
+      id-token: write
+      security-events: write
+    strategy:
+      matrix:
+        image:
+          [
+            "central-db",
+            "collector",
+            "collector-slim",
+            "main",
+            "roxctl",
+            "scanner",
+            "scanner-db",
+            "scanner-db-slim",
+            "scanner-slim",
+            "scanner-v4",
+            "scanner-v4-db",
+            "stackrox-operator",
+          ]
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-          ref: ${{ github.event.pull_request.head.sha }}
 
       - uses: ./.github/actions/job-preamble
         with:
           gcp-account: ${{ secrets.GCP_SERVICE_ACCOUNT_STACKROX_CI }}
 
-      - name: Checkout submodules
-        run: |
-            git submodule update --init
-
-      - uses: ./.github/actions/handle-tagged-build
-
-      - uses: actions/download-artifact@v4
+      - name: Central login
+        uses: stackrox/central-login@v1
         with:
-          name: cli-build
-
-      - name: Unpack cli build
-        run: |
-          tar xvzf cli-build.tgz
+          endpoint: ${{ vars.ACS_DOGFOODING_CENTRAL_URL }}
 
       - name: Install roxctl
-        run: |
-          make cli-install
-          roxctl version
+        uses: stackrox/roxctl-installer-action@v1
+        with:
+          central-endpoint: ${{ vars.ACS_DOGFOODING_CENTRAL_URL }}
+          central-token: ${{ env.ROX_API_TOKEN }}
 
-      - name: Scan images
+      - name: Scan images for vulnerabilities
         run: |
-          ./release/scripts/scan-images-with-roxctl.sh
+          release_tag=$(make tag)
+          roxctl image scan --output=sarif \
+            --image="quay.io/rhacs-eng/${{ matrix.image }}:${release_tag}" \
+            > results.sarif
```

Comment:
My only comment is that this would create a new security violation for each image push we are seeing since make tag will render to `4.4.x-529-g88b9dd155800`. This will create _a lot of_ noise and last time I checked there wasn't an API available to delete all violations.

Might make sense to use the pre-fixed tag here, i.e. only `4.4.x`?

Reply:
I think the de-duplication keys for the violations are the branch and the category, but not the image tag. So if a new commit `4.4.x-529-...` is pushed to master, we'll scan and upload violations for this image (e.g. `main`). At the same time, old vulnerabilities that are present in `main:4.4.x-528-...` but no longer in `main:4.4.x-529-...` are then automatically resolved.

But I can run some more tests on my fork to double check this. If you look on the security tab, the `resolved` vulnerabilities happened because initially I used the same category for all images, and then the images overwrote each other and resolved vulnerabilities because of that.

Reply:
I confirmed that uploading a security sarif file for `(category, branch)` will resolve all currently open vulnerabilities and replace them with the ones found in the file. This is regardless of the image tag. For example, after uploading `4.4.0` vulns after uploading `4.4.1` vulns for `category=main` and `branch=master` the security tab will only show vulnerabilities for `4.4.0` main image in `master`. In other words, this PR should give us the vulnerabilities found in the image built from the latest commit for each branch. Vulnerabilities in older commits will be auto resolved.